rust/lance-table/src/format/manifest.rs

```diff
@@ -189,9 +189,9 @@ impl Manifest {
         &self,
         ref_name: Option<String>,
         ref_path: String,
+        ref_base_id: u32,
```

Comment:
maybe we should call this `prepare_for_shallow_clone` since the result manifest is not complete yet.

Reply:
I guess it's fine then, maybe just add some explanations the behavior with doc string for the function should be good.